tests/testthat/test-github.R

```diff
@@ -0,0 +1,45 @@
+test_that("use_github_links populates empty URL field", {
+  create_local_package()
+  use_git()
```

Comment:
It seems like a lot to activate git in every test... I could instead mock `target_repo()`?

Reply:
You are going where no one's gone before re: testing these functions!

Yeah, if you can simplify the mocking that would be nice. This is why these functions haven't had tests before, which is also a position we could retreat to.

Reply:
I've taken your advice on using the various `skip_` and `local_` helpers. I'm not sure how to simplify the mocking other than making more helpers?

Reply:
I found the existing `github_url_from_git_remotes()`, which does the job of several lines, and is easy to mock https://github.com/r-lib/usethis/pull/1834/commits/db041b19074462ae56770148d0a5a0d2b2ca501f